examples/twittermap/web/public/javascripts/map/countmap.js

```diff
@@ -2,6 +2,208 @@ angular.module('cloudberry.map')
   .controller('countMapCtrl', function($scope, $compile, cloudberry, cloudberryConfig, MapResultCache,
                                        TimeSeriesCache, moduleManager, cloudberryClient, queryUtil) {
 
+    // Array to store the data for chart
+    $scope.chartData=[];
+    // Map to store the chart data for every polygon
+    $scope.ChartDataMap = new HashMap();
+    // The popup window shown now
+    $scope.popUp = null;
+
+    // return difference of two arrays
+    function arrayDiff (newArray, oldArray) {
+      var diffArray = [], difference = [];
+      for (var i = 0; i < newArray.length; i++) {
+        diffArray[newArray[i]] = true;
+      }
+      for (var j = 0; j < oldArray.length; j++) {
+        if (diffArray[oldArray[j]]) {
+          delete diffArray[oldArray[j]];
+        } else {
+          diffArray[oldArray[j]] = true;
+        }
+      }
+      for (var key in diffArray) {
+        difference.push(key);
+      }
+      return difference;
+    }
+
+    // Concat two hashmap results
+    function concatHashmap(newMap, cachedMap) {
+      if (cachedMap.count()===0) {
+        return newMap;
+      }
+
+      var concatMap = new HashMap();
+      newMap.forEach(function(value, key){
+        var concatValue = [];
+        var cacheValue = cachedMap.get(key);
+        if(value !== 0) {
+          if (cacheValue === undefined) {
+            concatValue=value;
+          }
+          else {
+            concatValue = value.concat(cacheValue);
+          }
+        }
+        else if (cacheValue !== undefined) {
+          concatValue=cacheValue;
+        }
+
+        concatMap.set(key,concatValue);
+      });
+      return concatMap;
+    }
+
+    // sum of one element in an array of objects
+    function sum(items, prop){
+      return items.reduce( function(previousVal, currentVal){
+        return previousVal + currentVal[prop];
+      }, 0);
+    }
+
+    function getPopupContent() {
+      // get chart data for the polygon
+      var geoIDChartData = $scope.ChartDataMap.get($scope.selectedGeoID);
+      (geoIDChartData && geoIDChartData.length!==0)? $scope.chartData = $scope.preProcess(geoIDChartData) : $scope.chartData = [];
```

Comment:
This way may be better for readability and simplicity:
`$scope.chartData = (geoIDChartData && geoIDChartData.lenght !== 0) ? $scope.preProcess(geoIDChartData) : []`